pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -200,3 +221,81 @@
 
 	Items []AzDriverNode `json:"items"`
 }
+
+type VolumeCapability struct {
+	// Specifies what API the volume will be accessed using. One of the
+	// following fields MUST be specified.
+	//
+	// Types that are valid to be assigned to AccessType:
+	//	*VolumeCapability_Block
+	//	*VolumeCapability_Mount
+	//TODO: figure out how to update the crd with this interface
+	AccessType isVolumeCapability_AccessType `json:"access_type"`
+	// This is a REQUIRED field.
+
+	AccessMode           *VolumeCapability_AccessMode `json:"access_mode,omitempty"`
+	XXX_NoUnkeyedLiteral struct{}                     `json:"-"`
+	XXX_unrecognized     []byte                       `json:"-"`
+	XXX_sizecache        int32                        `json:"-"`
+}
+
+//Specify how a volume can be accessed
+// +k8s:deepcopy-gen=false
+type VolumeCapability_AccessMode struct {
+	Mode VolumeCapability_AccessMode_Mode `json:"mode,omitempty"`
+}
+
+type VolumeCapability_AccessMode_Mode int32
+
+type isVolumeCapability_AccessType interface {
```

Comment:
You should omit the interfaces. Again, this is an implementation detail of gRPC.

Reply:
Is it fine to define the AccessType as an int the same way AccessMode is, or should it be excluded altogether? I'm not sure if it would ever be needed

Reply:
Yes